src/service/resources-impl.js

```diff
@@ -407,6 +410,12 @@ export class Resources {
    * @private
    */
   buildReadyResources_() {
+    // Avoid cases where elements add more elements inside of them
+    // and cause an infinite loop of building - see #3354 for details.
+    if (this.isCurrentlyBuildingPendingResources_) {
+      return;
+    }
+    this.isCurrentlyBuildingPendingResources_ = true;
```

Comment:
That sounds good. Does this mean if one resource keeps failing to build we're going to be stuck and not able to build all other pending resources? 

Should we also catch a failed build and remove resource from pending resources?